src/app/ClioApplication.cpp

```diff
@@ -102,9 +103,10 @@ ClioApplication::run(bool const useNgWebServer)
     auto whitelistHandler = web::dosguard::WhitelistHandler{config_};
     auto dosGuard = web::dosguard::DOSGuard{config_, whitelistHandler};
     auto sweepHandler = web::dosguard::IntervalSweepHandler{config_, ioc, dosGuard};
+    auto cache = data::LedgerCache{};
```

Comment:
It looks like an implementation leak. Maybe create a two overloads of `makeBackend()` with and without cache?

Reply:
What would it be instead? Either an instance inside of backend or a reference from outside? In my understanding backend should have never hosted the cache in the first place, it should have always been a dependency to inject. Can you explain your view?

Reply:
Hm, maybe it is ok for backend to own cache? Here we create cache, pass it to backend and never use it again. So why not place it into backend but with an option to mock?

Reply:
Initially i wanted the factory function to create the cache but that's not mockable and not really possible without using the heap. Cache is not some implementation detail - it's used across the system and should be injected in multiple components.. at the moment it's not this way simply because backend has the `cache` getters and we use `backend.cache()` in many places we should have injected the cache - this was always a mistake and a dirty hack that we wanted to get rid of. I don't think backend should own cache by any means.